app/views/event_attendances/index.html.erb

```diff
@@ -1,91 +1,123 @@
+    
+    <%- form_tag({}, :method => :get) do -%>
+        <p><%= label_tag :audience, "View Attendance For:" %>
+            <%= select_tag :audience,
+                            options_for_select(%w[Participant Student Mentor Volunteer].collect{|a| ["#{a} (#{@event.rsvpd_count(a)})", a]}, 
+                            @audience.to_s.classify),
+                            :onchange => "this.form.submit()" %>
+            <%= submit_tag "Go" %>
+        </p>
+    <% end -%>
+
+    <%- if @event.has_shifts?(@audience.to_s.classify) %>
+        <h3>Filter by Shift</h3>
+        
+        <div id="event_shift_selectors">
+        <%- for event_shift in @event.shifts.for(@audience.to_s.classify) -%>
+            <label for="event_shift_selector_<%= event_shift.id %>">
+                <input type="checkbox" id="event_shift_selector_<%= event_shift.id %>" data-shift-id="<%= event_shift.id %>" checked onChange="updateShiftSelections(this);">
+                <%= truncate(event_shift.title) %>
+            </label><br>
+        <% end -%>
+        <small class="light">Filtering the list will also update the email buttons above.</small>
+        </div>
+        
+    <% end -%>
 </div>
 
 <script type="text/javascript">
 
+function filterEvents() {
+
+}
+
+
 function updateShiftSelections(elem) {
-	var rows = $$("tr.event_attendance")
-	rows = rows.findAll(function(e) { return e.readAttribute('data-shift-id') == elem.readAttribute('data-shift-id') })
-	if(elem.checked == true) {
-		rows.invoke('show')
-	} else {
-		rows.invoke('hide')
-	}
-	updateEmailLinks()
+    // get the data shift id for the filter
+    var shiftid = elem.getAttribute("data-shift-id");
+
+    // get all rows that correspond to that shift id
+    var rows = $("tr.event_attendance[data-shift-id='" + shiftid + "']");
+
+    if (elem.checked)  {
+        rows.show();
+    } else {
+        rows.hide();
+    }
+    
+    updateEmailLinks();
 }
 
 function updateEmailLinks() {
-	var rows = $$("tr.event_attendance").findAll(function(e) { return e.visible() })
-	var attending_rows = rows.findAll(function(e) { return e.readAttribute("data-attending") == "true" })
-	var attended_rows =  rows.findAll(function(e) { return e.readAttribute("data-attended") == "true" })
-	$('attending_email_link').writeAttribute("href", "mailto:" + attending_rows.collect(function(e) { return e.readAttribute("data-email") }).join(","))
-	$('attended_email_link').writeAttribute("href",  "mailto:" + attended_rows.collect(function(e) {  return e.readAttribute("data-email") }).join(","))
+    // get rows which are visible and corresponding attributes are true
+    var attending_rows = $("tr.event_attendance[data-attending='true']:visible");
+    var attended_rows = $("tr.event_attendance[data-attended='true']:visible");
+
+    // create an email list of those people attending
+    var emailsAttending = [];
+    function getAttending() {
```

Comment:
Does this function get called anywhere else, or could it just be removed from the function wrapper?